packages/state-transition/src/util/epochShuffling.ts

```diff
@@ -102,42 +103,67 @@ export function computeCommitteeCount(activeValidatorCount: number): number {
   return Math.max(1, Math.min(MAX_COMMITTEES_PER_SLOT, committeesPerSlot));
 }
 
-export function computeEpochShuffling(
-  state: BeaconStateAllForks,
-  activeIndices: Uint32Array,
-  epoch: Epoch
-): EpochShuffling {
-  const activeValidatorCount = activeIndices.length;
-
-  const shuffling = activeIndices.slice();
-  const seed = getSeed(state, epoch, DOMAIN_BEACON_ATTESTER);
-  unshuffleList(shuffling, seed);
-
+function buildCommitteesFromShuffling(
+  activeValidatorCount: number,
+  shuffling: Uint32Array
+): Pick<EpochShuffling, "committees" | "committeesPerSlot"> {
   const committeesPerSlot = computeCommitteeCount(activeValidatorCount);
-
   const committeeCount = committeesPerSlot * SLOTS_PER_EPOCH;
 
-  const committees: Uint32Array[][] = [];
+  const committees = new Array<Uint32Array[]>(SLOTS_PER_EPOCH);
   for (let slot = 0; slot < SLOTS_PER_EPOCH; slot++) {
-    const slotCommittees: Uint32Array[] = [];
+    const slotCommittees = new Array<Uint32Array>(committeesPerSlot);
+
     for (let committeeIndex = 0; committeeIndex < committeesPerSlot; committeeIndex++) {
       const index = slot * committeesPerSlot + committeeIndex;
       const startOffset = Math.floor((activeValidatorCount * index) / committeeCount);
       const endOffset = Math.floor((activeValidatorCount * (index + 1)) / committeeCount);
       if (!(startOffset <= endOffset)) {
         throw new Error(`Invalid offsets: start ${startOffset} must be less than or equal end ${endOffset}`);
       }
-      slotCommittees.push(shuffling.subarray(startOffset, endOffset));
+      slotCommittees[committeeIndex] = shuffling.subarray(startOffset, endOffset);
     }
-    committees.push(slotCommittees);
+
+    committees[slot] = slotCommittees;
   }
 
+  return {
+    committees,
+    committeesPerSlot,
+  };
+}
+
+export function computeEpochShuffling(
+  state: BeaconStateAllForks,
+  activeIndices: Uint32Array,
+  epoch: Epoch
+): EpochShuffling {
+  const activeValidatorCount = activeIndices.length;
```

Comment:
you should be able to refactor `buildCommitteesFromShuffling` to only take the `shuffling` and compute `activeValidatorCount` inside that fn: `const activeValidatorCount = shuffling.length;`

You also might consider removing `committeesPerSlot` from the return value and simply (re)compute it.

Ultimately this function can look like:
```ts
export function computeEpochShuffling(
  state: BeaconStateAllForks,
  activeIndices: Uint32Array,
  epoch: Epoch
): EpochShuffling {
  const seed = getSeed(state, epoch, DOMAIN_BEACON_ATTESTER);
  const shuffling = unshuffleList(activeIndices, seed, SHUFFLE_ROUND_COUNT);
  const committees = buildCommitteesFromShuffling(shuffling);
  const committeesPerSlot = committees[0].length;
  return {
    epoch,
    activeIndices,
    shuffling,
    committees,
    committeesPerSlot,
  };
}
```

It may be nice to remove `state` from these functions (eg: `computeEpochShuffling(activeIndices: Uint32Array, seed: Uint8Array): EpochShuffling`), but thats probably a bit too much for this PR.

Reply:
Much cleaner.  Updated.

Reply:
> It may be nice to remove state from these functions (eg: computeEpochShuffling(activeIndices: Uint32Array, seed: Uint8Array): EpochShuffling), but thats probably a bit too much for this PR.

Agreed.  the seed would need to be passed in and there are a bunch of locations where this is called.  can make it a TODO for later though